### DISCUSSION TITLE
Package Idea: Outlined Geoms

### DISCUSSION THREAD

**Discussion by teunbrand** [ggplot2-extenders/ggplot-extension-club#2] [Ideas]
So, I had an idea for an extension package that could leverage the new [R 4.2.0 features](https://www.stat.auckland.ac.nz/~paul/Reports/GraphicsEngine/paths/paths.html) to make outlines of arbitrary geoms and theme elements.

There are already various endeavours with a similar aim, notably:
- [shadowtext](https://github.com/GuangchuangYu/shadowtext)
- [ggtrace](https://github.com/rnabioco/ggtrace) (yes June, the *other* ggtrace)
- My own [geom_outline_point](https://teunbrand.github.io/ggh4x/reference/geom_outline_point.html
- Feel free to comment more of such packages

But all of the above don't use the new features in R. In addition to gathering a lot of outlining tools in one place and using the new features, it would likely also be possible to outline boxplots for example. Obvious downside at this time is that not all graphics devices support the new features yet, such as {ragg} or {svglite}. Any thoughts?

**Comment by EvaMaeRey** [comment]
Sounds useful!  I've recently wanted to outline a line!  I feel like I saw someone doing this somewhere - w/ 'once you start outlining your lines you'll never go back' messaging, but I can't remember where, when.

**Reply by teunbrand** [reply]
I have made a draft here: https://github.com/teunbrand/ggoutlines.
Not sure how that has hold up over new ggplot2 releases and such

**Reply by EvaMaeRey** [reply]
exciting - very 3D feels!  😮

**Reply by EvaMaeRey** [reply]
Wanted to use in this context where sometimes you need a bar instead of a circle; and then you could use GeomLine, with lineend = "round"  but you don't have stroke for outlining as with GeomPoint.  A different way would be to compute polygons...  

![image](https://github.com/user-attachments/assets/359c41b3-b07e-45f2-bb94-9aae5e20d8ae)

**Reply by yjunechoe** [reply]
> GeomLine, with lineend = "round" but you don't have stroke for outlining

I'm pretty sure [ggchicket](https://github.com/hrbrmstr/ggchicklet) can do this!

**Reply by EvaMaeRey** [reply]
Oh!  I definitely need to check out those chicklet Geoms!   Rounded rectangle sounds about right!

**Reply by teunbrand** [reply]
I also made a rounded rectangle geom (or at least, it allows for rounded rectangles): https://teunbrand.github.io/ggh4x/reference/geom_box.html. I sometimes have the feeling that people (myself included) keep reinventing the wheel on some geoms.

**Reply by EvaMaeRey** [reply]
I will check this out geom_box too - I remembered geom_box but not it's roundness capabilities.  

For using GeomLine roundness gave a radius is 1/2 its width and the roundness extended beyond its length. (so zero length line would still give you a circle!)  I felt like that was going to be kind of easy to think about and work with, but lacking an outline is downside.  Much to weigh, when coming back to this project. 🙃
